contrib/cirrus/runner.sh

```diff
@@ -345,6 +364,51 @@ dotest() {
         |& logformatter
 }
 
+# Optimization: will exit if the only PR diffs are under docs/ or tests/
+# with the exception of any given arguments. E.g., don't run e2e or upgrade
+# or bud tests if the only PR changes are in test/system.
+function _bail_if_test_can_be_skipped() {
+    local head base diffs
+
+    # Cirrus sets these for PRs but not cron. In cron, we never want to skip.
+    for v in CIRRUS_CHANGE_IN_REPO DEST_BRANCH; do
+        if [[ -z "${!v}" ]]; then
+            echo "[ _cannot do selective skip: \$$v is undefined ]"
```

Comment:
Non-blocking: Would prefer use of `msg()` and/or `warn()` functions for consistency (warn() prints the LOC it was called from).